stdlib/public/core/MigrationSupport.swift

```diff
@@ -700,3 +700,13 @@ public typealias _DefaultCustomPlaygroundQuickLookable = __DefaultCustomPlaygrou
 public protocol __DefaultCustomPlaygroundQuickLookable {
   var _defaultCustomPlaygroundQuickLook: _PlaygroundQuickLook { get }
 }
+
+extension String {
+  /// A type that represents the number of steps between two `String.Index`
+  /// values, where one value is reachable from the other.
+  ///
+  /// In Swift, *reachability* refers to the ability to produce one value from
+  /// the other through zero or more applications of `index(after:)`.
+  @available(*, deprecated, message: "All index distances are now of type Int")
+  public typealias IndexDistance = Int
```

Comment:
Typealiases defined in protocol extensions won't work in `where` clauses. If you care, you should leave it in the protocol itself.

Reply:
I believe this is fine — this is a typealias in the concrete `String` type, not the one in `Collection`. (The protocol typealias remains in place.)